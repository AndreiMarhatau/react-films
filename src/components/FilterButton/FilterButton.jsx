import React from 'react';
import styles from './FilterButton.scss';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const FilterButton = (props) => {
    return <div className={styles.container}>
        <a href='' className={classNames(styles.button, props.isActive ? styles.active : null)} onClick={props.clickHandler}>
          {props.text}
        </a>
        {props.isActive && <hr className={styles.hr}/>}
      </div>;
}

FilterButton.propTypes = {
  isActive: PropTypes.bool,
  text: PropTypes.string,
  clickHandler: PropTypes.func
};

export default FilterButton;