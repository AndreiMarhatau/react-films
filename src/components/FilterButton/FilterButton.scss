.button{
  color: white;
  font-size: 22px;
}
.active{

}

.container{
  position: relative;
}

.hr{
  position: absolute;
  width: 100%;
  height: 1px;
  color: #f65261;
  background-color: #f65261;
}