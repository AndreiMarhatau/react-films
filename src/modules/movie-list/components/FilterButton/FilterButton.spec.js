import React from 'react';
import {render} from '@testing-library/react';

import FilterButton from './FilterButton';

describe('FilterButton', () => {
    test('FilterButton shapshot', () => {
        const {asFragment} = render(<FilterButton/>);
        expect(asFragment(<FilterButton/>)).toMatchSnapshot();
    })
})