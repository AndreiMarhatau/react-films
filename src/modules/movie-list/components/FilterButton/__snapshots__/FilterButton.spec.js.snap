// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton FilterButton shapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <a
      class="button"
      href=""
    />
  </div>
</DocumentFragment>
`;
